From 14aeb21617350b53c199af38fca8d1ebe55b880b Mon Sep 17 00:00:00 2001
From: David Disseldorp <ddiss@suse.de>
Date: Tue, 1 Sep 2015 19:50:50 +0200
Subject: [PATCH] target/rbd: add stubbed out pr_read_full_status
Patch-mainline: Not yet, SES2 clustered LIO/RBD
References: fate#318836

Add a tcm_rbd_pr_ops handler for PR READ FULL STATUS requests. This
operation is currently unsupported by the RBD backend.

Signed-off-by: David Disseldorp <ddiss@suse.de>

---
 drivers/target/target_core_rbd.c |    9 +++++++++
 1 file changed, 9 insertions(+)

--- a/drivers/target/target_core_rbd.c
+++ b/drivers/target/target_core_rbd.c
@@ -1729,6 +1729,14 @@ tcm_rbd_execute_pr_report_capabilities(s
 	return TCM_NO_SENSE;
 }
 
+static sense_reason_t
+tcm_rbd_execute_pr_read_full_status(struct se_cmd *cmd, unsigned char *buf,
+				    u32 buf_len)
+{
+	pr_err("READ FULL STATUS not supported by RBD backend\n");
+	return TCM_LOGICAL_UNIT_COMMUNICATION_FAILURE;
+}
+
 /* handle PR registration for a currently unregistered I_T nexus */
 static sense_reason_t
 tcm_rbd_execute_pr_register_new(struct tcm_rbd_pr_info *pr_info, u64 old_key,
@@ -2516,6 +2524,7 @@ static struct target_pr_ops tcm_rbd_pr_o
 	.pr_read_keys		= tcm_rbd_execute_pr_read_keys,
 	.pr_read_reservation	= tcm_rbd_execute_pr_read_reservation,
 	.pr_report_capabilities	= tcm_rbd_execute_pr_report_capabilities,
+	.pr_read_full_status	= tcm_rbd_execute_pr_read_full_status,
 
 	.pr_register		= tcm_rbd_execute_pr_register,
 	.pr_reserve		= tcm_rbd_execute_pr_reserve,
